### PR TITLE
Parse description in while init RSSItem

### PR DESCRIPTION
**Pull request by Chenyuyao**

### CODE REVIEW

File: motcha/Storage/MCCoreDataWordOccurrence.m [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
+//  MCCoreDataWordOccurrence.m
+//  motcha
+//
+//  Created by Kevin on 2015-03-25.
+//  Copyright (c) 2015 Frank Li. All rights reserved.
+//
```
Comment (shinfan) [inline motcha/Storage/MCCoreDataWordOccurrence.m]: Remove

File: motcha/Storage/MCCoreDataWordOccurrence.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+//
+//  MCCoreDataWordOccurrence.h
+//  motcha
+//
+//  Created by Kevin on 2015-03-25.
+//  Copyright (c) 2015 Frank Li. All rights reserved.
+//
```
Comment (shinfan) [inline motcha/Storage/MCCoreDataWordOccurrence.h]: remove

File: motcha/Storage/MCDatabaseManager.h [hunk 3]
```diff
@@ -23,4 +23,11 @@
                        onPredicate:(NSPredicate *)predicate
                    completionBlock:(void(^)(NSError *error))block;
 
+//This method should only be called in background thread
+- (NSArray *)fetchEntriesForEntityName:(NSString *)name
```
Comment (shinfan) [inline motcha/Storage/MCDatabaseManager.h]: Don't make this public. Use the async one

File: motcha/RecommendationALG/MCTitlePipe.m [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    MCCoreDataDictionaryWord * coreDataDictionaryWord =
+    (MCCoreDataDictionaryWord *)[dictionaryWords objectAtIndex:0];
+    wordScore = [[MCTitlePipe getPosScore] objectForKey:[coreDataDictionaryWord pos]];
+    if (wordScore == nil) {
+      wordScore = @0;
+    }
+  }
+  
+  //Get the word count and last modified date from Core Data
+  NSNumber * occurrence = @0;
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.m]: @(0)

File: motcha/RecommendationALG/MCTitlePipe.m [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+//Store word in MCWordOccurrence Data Model
+- (void) saveWord:(NSString *) word {
+  //Find word through Core Data
+  //Fetch Request to find the target word
+  NSPredicate * predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"word", word];
+  NSError * error = nil;
+  NSArray *wordOccurrences =
+  [[MCDatabaseManager defaultManager] fetchEntriesForEntityName:kStrWordOccurrenceEntityName
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.m]: You run this on a main thread it's gonna block UI, use async

File: motcha/Storage/MCDatabaseManager.h (hunk 3 above)
Comment (franklixuefei) [inline motcha/Storage/MCDatabaseManager.h]: shin我改了你的code 现在可以specify 是否想要async 因为有些地方需要sync的version

- (void)fetchEntriesForEntityName:(NSString *)name
                            async:(BOOL)shouldExecuteAsync
                      onPredicate:(NSPredicate *)predicate
                           onSort:(NSArray *)sortDescriptors
                  completionBlock:(void(^)(NSArray *, NSError *))block;


- (void)deleteEntriesForEntityName:(NSString *)name
                             async:(BOOL)shouldExecuteAsync
                       onPredicate:(NSPredicate *)predicate
                   completionBlock:(void(^)(NSError *error))block;

File: motcha/RecommendationALG/MCTitlePipe.m [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    }
+  } else {
+    NSLog(@"Error fetch word occurrence in MCTitlePipe.getWordScore");
+    NSLog(@"%@ %@", error, error.localizedDescription);
+  }
+
+}
+
+//Return the Recommended ALG Score for an array of words
+- (NSNumber *) wordScoreSink:(NSArray *) contentwordArray {
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.m]: Why you always use NSNumber which is a pointer? CGFloat or double could do the job.

File: motcha/Storage/MCDatabaseManager.h (hunk 3 above)
Comment (franklixuefei) [inline motcha/Storage/MCDatabaseManager.h]: kevin这个pull request里面已经用了这个change

File: motcha/RecommendationALG/MCTitlePipe.m [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+//Save all word in array into Core Data
+- (void) saveWordSink:(NSArray *) contentwordArray {
+  for (NSString * word in contentwordArray) {
+    [self saveWord:word];
+  }
+}
+
+//get the total score of a title
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.m]: You are just repeating the function name in your comment

File: motcha/RecommendationALG/MCTitlePipe.m [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+  NSError * error = nil;
+  NSArray *wordOccurrences =
+  [[MCDatabaseManager defaultManager] fetchEntriesForEntityName:kStrWordOccurrenceEntityName
+                                                    onPredicate:predicate
+                                                         onSort:nil
+                                                          error:nil];
+  if (!error) {
+    if ([wordOccurrences count] == 0) {
+      //word does not in MCWordOccurrence Data Model, create an instance with count = 0
+      MCCoreDataWordOccurrence * newWord =
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.m]: You should not use the database manager here directly, you should wrap it with the localStorageService.
But we can fix this later.

File: motcha/RecommendationALG/MCTitlePipe.m [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+  
+  for (NSUInteger index = 0; index < [contentWordArray count]; index++)
+  {
+    NSString* currentWord = [contentWordArray objectAtIndex:index];
+    NSPredicate * predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"word", currentWord];
+    NSArray * dictionaryWords = [[MCDatabaseManager defaultManager] fetchEntriesForEntityName:kStrDictionaryEntityName onPredicate:predicate onSort:nil error:nil];
+    if ([dictionaryWords count] == 1) {
+      MCCoreDataDictionaryWord * coreDataDictionaryWord =
+      (MCCoreDataDictionaryWord *)[dictionaryWords objectAtIndex:0];
+      if ([[MCTitlePipe getFunctionWords] containsObject:[coreDataDictionaryWord pos]]) {
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.m]: functionwords should be a NSSet, not NSArray.

File: motcha/RecommendationALG/MCTitlePipe.h [hunk 10]
```diff
@@ -0,0 +1,14 @@
+#import <Foundation/Foundation.h>
+
+//A pipeline to process RSS feed title
+//MCTitlePipe is designed as Pipe and Filter Architecture
+@interface MCTitlePipe: NSObject
+
+
+//Methods there takes some time to excute and should be put in background queue
```
Comment (shinfan) [inline motcha/RecommendationALG/MCTitlePipe.h]: database manager has a dedicated queue to execute database IO, we should rely on that. We should never handle multithread on the higher level. And if it is async, we need a callback block for each method.

File: motcha/Storage/MCDatabaseManager.h (hunk 3 above)
Comment (shinfan) [inline motcha/Storage/MCDatabaseManager.h]: kevin得这个save word要用async, and actually here is the only place we need to use async.

**Pull request closed without merging**